extract.py

```diff
@@ -120,6 +143,87 @@ def extract_database(args):
     return output_rows
 
 
+def extract_csv(args, mapping=None, init_id=0, date_format="%y%m%d", reg_rules=None):
+    if mapping is None:
+        mapping = dict()
+    if reg_rules is None:
+        reg_rules = dict()
+    translation = {"meta": {"reg_rules": reg_rules}, "mapping": mapping}
+    translation["meta"]["date_format"] = date_format
+    output_rows = []
+    report = get_report()
+    person_id = init_id
+    with open(args.source, "r") as datasource:
+        rows = csv.DictReader(datasource)
+        for row in rows:
+            handled_row = translate_row(row, report, translation)
+            if not handled_row[0]:
+                handled_row[0] = person_id
+                person_id += 1
+            output_rows.append(handled_row)
+
+    shuffle(output_rows)
+    if args.verbose:
+        print_report(report)
+    return output_rows
+
+
+def translate_row(row, report, translation):
+    output_row = []
+    reg_rules = translation["meta"]["reg_rules"]
+    mapping = translation["mapping"]
+
+    record_id = case_insensitive_lookup(row, "record_id", mapping)
+    output_row.append(record_id)
+
+    given_name = case_insensitive_lookup(row, "given_name", mapping)
+    validate(report, "given_name", given_name, reg_rules)
+    clean_given_name = clean_string(given_name)
+    output_row.append(
+        reg_rules.get("given_name", {}).get(clean_given_name, clean_given_name)
+    )
+
+    family_name = case_insensitive_lookup(row, "family_name", mapping)
+    validate(report, "family_name", family_name, reg_rules)
+    clean_family_name = clean_string(family_name)
+    output_row.append(
+        reg_rules.get("family_name", {}).get(clean_family_name, clean_family_name)
+    )
+
+    dob = case_insensitive_lookup(row, "DOB", mapping)
+    dob = clean_dob_fromstr(dob, translation["meta"]["date_format"])
+    validate(report, "DOB", dob, reg_rules)
+    output_row.append(reg_rules.get("DOB", {}).get(dob, dob))
+
+    sex = case_insensitive_lookup(row, "sex", mapping)
+    validate(report, "sex", sex, reg_rules)
+    output_row.append(reg_rules.get("sex", {}).get(sex, sex))
+
+    # is phone or phone_number the canonical field name?
+    phone_number = case_insensitive_lookup(row, "phone", mapping)
+    validate(report, "phone_number", phone_number, reg_rules)
```

Comment:
Should this be `phone`?

```suggestion
    validate(report, "phone", phone_number, reg_rules)
```